test/jdk/java/util/Calendar/NarrowNamesTest.java

```diff
@@ -95,26 +96,27 @@ public static void main(String[] args) {
                 );
         testMap(US, DAY_OF_WEEK, NARROW_FORMAT); // expect null
         if (providers.startsWith("CLDR")) {
-            testMap(US, AM_PM, ALL_STYLES,
-                    "AM",
-                    "PM",
-                    "midnight",
-                    "noon",
-                    "in the morning",
-                    "",
-                    "in the afternoon",
-                    "",
-                    "in the evening",
-                    "",
-                    "at night",
-                    "",
-                    RESET_INDEX,
-                    "a", "p", "mi", "n", "", "", "", "", "", "", "", "");
+            Map<String, Integer> expectedFieldValues = new HashMap<>();
+            expectedFieldValues.put("AM", Calendar.AM);
+            expectedFieldValues.put("PM", Calendar.PM);
+            expectedFieldValues.put("midnight", null);
+            expectedFieldValues.put("noon", null);
+            expectedFieldValues.put("in the morning", null);
+            expectedFieldValues.put("in the afternoon", null);
+            expectedFieldValues.put("in the evening", null);
+            expectedFieldValues.put("at night", null);
+            expectedFieldValues.put("a", Calendar.AM);
+            expectedFieldValues.put("p", Calendar.PM);
+            expectedFieldValues.put("mi", null);
+            expectedFieldValues.put("n", null);
+            testAM_PM(US, ALL_STYLES, expectedFieldValues);
         } else {
-            testMap(US, AM_PM, ALL_STYLES,
-                    "AM", "PM",
-                    RESET_INDEX,
-                    "a", "p");
+            Map<String, Integer> expectedFieldValues = new HashMap<>();
+            expectedFieldValues.put("AM", Calendar.AM);
+            expectedFieldValues.put("PM", Calendar.PM);
+            expectedFieldValues.put("a", Calendar.AM);
+            expectedFieldValues.put("p", Calendar.PM);
+            testAM_PM(US, ALL_STYLES, expectedFieldValues);
```

Comment:
I believe this can be reverted to the original, as the original code compares the exact map (including the length of the map).  , i.e., 
```
           testMap(US, AM_PM, ALL_STYLES,
                    "AM", "PM",
                    RESET_INDEX,
                    "a", "p");
```

Reply:
Done. Reverted this specific part to what it was originally. Test continues to pass with this change and the latest PR update.